### DISCUSSION TITLE
Support for Ollama

### DISCUSSION THREAD

**Discussion by maalik-hrm** [didalgolab/ai-intellij-plugin#22] [Ideas]
First of all, this is a great plugin!

It would be great if there was an option to add a custom LLM configuration for a Ollama server. I guess the only things that needs to be done in order to make this work is the following:
- In the settings add a new entry under GPT-3.5 Turbo/GPT-4 called Custom Model
- The setting page could look like this:
<img width="719" alt="Screenshot 2024-05-24 at 18 45 36" src="https://github.com/didalgolab/chatgpt-intellij-plugin/assets/122912202/00ee63d0-fbd6-41b8-b6ae-446f0094112d">

- It should be possible to add custom Models to the Model dropdown, or change this to a textfield

Here is a Blogpost from ollama about the compatibility with OpenAI: [https://ollama.com/blog/openai-compatibility](https://ollama.com/blog/openai-compatibility)

Keep up the great work and have a nice day!

**Comment by didalgolab** [comment]
Thank you for this suggestion. I'm currently working on the next release with expanded support for other API providers, besides OpenAI, and I plan to add Ollama support too. Stay tuned!

**Comment by didalgolab** [comment]
Support for local Ollama models was added in v1.0.0. It's disabled by default, may be enabled in the configuration settings:

- visit `Ollama` configuration page
- select `Enabled in a tool window` option
- provide a model name

![obraz](https://github.com/didalgolab/chatgpt-intellij-plugin/assets/121457297/11558665-46f2-434c-be14-2bf173dd3fb6)

Feel free to try it out once the release is on the Marketplace and provide any feedback you might have.
